torch/_inductor/codegen/mps.py

```diff
@@ -0,0 +1,170 @@
+        return f"metal::abs({x})"
+
+    @staticmethod
+    def sin(x):
+        return f"metal::sin({x})"
+
+    @staticmethod
+    def cos(x):
+        return f"metal::cos({x})"
+
+    @staticmethod
+    def tan(x):
+        return f"metal::tan({x})"
+
+    @staticmethod
+    def asin(x):
+        return f"metal::asin({x})"
+
+    @staticmethod
+    def acos(x):
+        return f"metal::acos({x})"
+
+    @staticmethod
+    def atan(x):
+        return f"metal::atan({x})"
+
+    @staticmethod
+    def sqrt(x):
+        return f"metal::sqrt({x})"
+
+
+class MetalKernel(SIMDKernel):
+    overrides = MetalOverrides  # type: ignore[assignment]
+    suffix = ";"
+    newvar_prefix = "auto "
+
+    def __init__(
+        self,
+        tiling: dict[str, sympy.Expr],
+        **kwargs,
+    ) -> None:
+        super().__init__(tiling, **kwargs)
+        self.compute = self.body
+        self.loads = self.body
+        self.stores = self.body
+
+    def dtype_to_str(self, dtype: torch.dtype) -> str:
+        return DTYPE_TO_METAL[dtype]
+
+    def load(self, name: str, index: sympy.Expr):
+        """Codegen a load from an InputBuffer"""
+        var = self.args.input(name)
+        index = self.prepare_indexing(index)
+        line = f"{var}[{index}]"
+        return self.cse.generate(self.body, line)
+
+    def store(
+        self, name: str, index: sympy.Expr, value: CSEVariable, mode: StoreMode = None
+    ) -> None:
+        var = self.args.output(name)
+        index = self.prepare_indexing(index)
+        dtype_str = self.dtype_to_str(V.graph.get_dtype(name))
+        line = f"{var}[{index}] = static_cast<{dtype_str}>({value});"
+        self.body.writeline(DeferredLine(name, line))
+
+    def codegen_kernel(self, name=None):
+        """Called at the end to generate a final kernel string"""
+        code = IndentedBuffer()
+        code.writeline('torch.mps._compile_shader("""')
+        with code.indent():
+            code.writeline("kernel void kernel_0(")
+            with code.indent():
+                for outer, inner in self.args.output_buffers.items():
+                    dtype_str = self.dtype_to_str(V.graph.get_dtype(outer))
+                    code.writeline(f"device {dtype_str}* {inner},")
+                for outer, inner in self.args.input_buffers.items():
+                    dtype_str = self.dtype_to_str(V.graph.get_dtype(outer))
+                    code.writeline(f"constant {dtype_str}* {inner},")
+                code.writeline("uint x0 [[thread_position_in_grid]]")
+            code.writeline(") {")
+            with code.indent():
+                code.splice(self.body)
+            code.writeline("}")
+        code.writeline('""")')
+
+        return code.getvalue()
+
+    def call_kernel(self, name: str, node=None):
+        """Codegen a call to this kernel"""
+        wrapper = V.graph.wrapper_code
+        args = list(self.args.output_buffers.keys()) + list(
+            self.args.input_buffers.keys()
+        )
```

Comment:
```suggestion
        args = [*self.args.output_buffers.keys(), *self.args.input_buffers.keys()]
```

Do you need to handle inplaced buffers and sizes here?